action/protocol/staking/bucket_statemanager.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/state"
+	"github.com/pkg/errors"
+)
+
+type (
+	bucketStateManager struct {
+		protocol.StateManager
+		config bucketStateConfig
+	}
+
+	bucketStateConfig struct {
+		bucketNamespace  string
+		totalBucketKey   []byte
+		bucketKeygenFunc func(*VoteBucket) []byte
+	}
+)
+
+var _executorBucketStateConfig = bucketStateConfig{
+	bucketNamespace: _executionStakingNameSpace,
+	totalBucketKey:  TotalBucketKey,
+	bucketKeygenFunc: func(bucket *VoteBucket) []byte {
+		return bucketKey(bucket.Index)
+	},
+}
+
+func newExecutorBucketStateManager(sm protocol.StateManager) *bucketStateManager {
```

Comment:
since the `new()` does not take a `config` as input, it's always using the const/var `_executorBucketStateConfig`, so use of this `config` is not much necessary